src/main/java/com/opentable/db/postgres/embedded/EmbeddedPostgres.java

```diff
@@ -707,16 +737,12 @@ private static File prepareBinaries(PgBinaryResolver pgBinaryResolver)
                 throw new IllegalStateException("No Postgres binary found for " + system + " / " + machineHardware);
             }
 
-            try (DigestInputStream pgArchiveData = new DigestInputStream(
-                        pgBinary, MessageDigest.getInstance("MD5"));
-                FileOutputStream os = new FileOutputStream(pgTbz))
-            {
-                IOUtils.copy(pgArchiveData, os);
+            try (DigestInputStream pgArchiveData = new DigestInputStream(pgBinary, MessageDigest.getInstance("MD5"));
+                 ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
```

Comment:
This loads everything into memory. That seems a bad thing

Reply:
Writing to a file has been already used. With this change, I tries to avoid unnecessary disk operations. The size of the file being processed is about 15 MB, which I think is not so much. So either I can revert this change or I would suggest two-round processing. In the first round only calculate the hash and in the second round extract the file. The `PgBinaryResolver#getPgBinary` method for obtaining a stream would be called twice.